src/main/webapp/app/exam/manage/student-exams/student-exam-timeline/student-exam-timeline.component.ts

```diff
@@ -0,0 +1,373 @@
+     * helper method to check if the object is a file upload submission
+     * @param object the object to check
+     */
+    private isFileUploadSubmission(object: FileUploadSubmission | SubmissionVersion | ProgrammingSubmission | undefined) {
+        if (!object) {
+            return false;
+        }
+        const fileUploadSubmission = object as FileUploadSubmission;
+        return !!fileUploadSubmission.id && fileUploadSubmission.submissionDate && fileUploadSubmission.filePath;
+    }
+
+    /**
+     * Finds the submission for the exercise with the closest timestamp to the current timestamp.
+     * @param exercise The exercise for which the submission should be found.
+     */
+    private findSubmissionForExerciseClosestToCurrentTimeStampForExercise(exercise: Exercise) {
+        const comparisonObject = dayjs(this.value);
+        let smallestDiff = Number.MAX_VALUE;
+        let timestampWithSmallestDiff = 0;
+        if (exercise.type === ExerciseType.PROGRAMMING) {
+            timestampWithSmallestDiff = this.findClosestTimestampForExerciseInSubmissionArray(exercise, this.programmingSubmissions);
+        } else if (exercise.type === ExerciseType.FILE_UPLOAD) {
+            // file upload exercises only have one submission
+            return this.fileUploadSubmissions.find((submission) => submission.participation?.exercise?.id === exercise.id);
+        } else {
+            const numberOfSubmissionsForExercise = this.submissionVersions.filter((submission) => submission.submission?.participation?.exercise?.id === exercise.id).length;
+            for (let i = 0; i < this.submissionVersions.length; i++) {
+                if (
+                    Math.abs(this.submissionVersions[i].createdDate.diff(comparisonObject)) < smallestDiff &&
+                    this.submissionVersions[i]?.submission.participation?.exercise?.id === exercise.id &&
+                    (!this.submissionVersions[i].createdDate.isSame(comparisonObject) || numberOfSubmissionsForExercise === 1)
+                ) {
+                    smallestDiff = Math.abs(this.submissionVersions[i].createdDate.diff(comparisonObject));
+                    timestampWithSmallestDiff = this.submissionVersions[i].createdDate.valueOf();
+                }
+            }
+        }
+        return this.findCorrespondingSubmissionForTimestamp(timestampWithSmallestDiff);
+    }
+
+    /**
+     * Finds the closest timestamp for a submission of a given exercise in a given submission array
+     * @param exercise the exercise for which the submission should be found
+     * @param submissions the submissions array in which the submission should be found
+     */
+    private findClosestTimestampForExerciseInSubmissionArray(exercise: Exercise, submissions: ProgrammingSubmission[] | FileUploadSubmission[]): number {
+        const comparisonObject = dayjs(this.value);
+        let smallestDiff = Number.MAX_VALUE;
+        let timestampWithSmallestDiff = 0;
+        // @ts-ignore
```

Comment:
What do you have to ignore here?

Reply:
i resolved it now by specifing the array as array of submissions instead of array of fil upload submissions or array of programming submissions since it doesn't matte here.